programs/ssl/ssl_server2.c

```diff
@@ -3884,7 +3884,17 @@ int main( int argc, char *argv[] )
                     mbedtls_printf( "\n    " );
                 mbedtls_printf( "%02x ", dtls_srtp_key_material[j] );
             }
+            mbedtls_printf( "\n" );
 
+            /* produce a less readable output used to perform automatic checks
+             * - compare client and server output
+             * - interop test with openssl which client produces this kind of output
+             */
+            mbedtls_printf( "    Keying material: " );
```

Comment:
Not a blocker: I wonder if  it's worth making the new `-g` option a bit more flexible so that it's not necessary for server and client output to use _exactly_ the same output format. For example, `-g` might take two functions as arguments which are run on server/client output and whose output is then compared.  @mpg @gilles-peskine-arm ?

However, I think we should wait until the second review before changing anything - the current solution works, and the duplicated output isn't a major nuisance.

Reply:
I feel YAGNI here. This is an internal interface and I don't expect a lot of pull requests in flight using the option, so we can change the semantics whenever we want.